### ISSUE TITLE
Object of type AlertField is not JSON serializable

### ISSUE THREAD

**Issue by markusos** [looker-open-source/sdk-codegen#1104]
After a lot of back and forth with the looker support we where directed to instead submit an issue here instead.

This is a regression issue for the SDK, the following code works in ```"looker-sdk" = "21.20.0"``` but not in the latest release: ```"looker-sdk" = "22.4.0"```

Sample code to reproduce: 
```
import os

import looker_sdk
from looker_sdk import models40  # using Looker 4.0

os.environ.update(
    {
        "LOOKERSDK_CLIENT_ID": "<LOOKERSDK_CLIENT_ID>",
        "LOOKERSDK_CLIENT_SECRET": "<LOOKERSDK_CLIENT_SECRET>",
        "LOOKERSDK_BASE_URL": "<LOOKERSDK_BASE_URL>",
    }
)

sdk = looker_sdk.init40()

write_alert = models40.WriteAlert(
    applied_dashboard_filters=[
        models40.AlertAppliedDashboardFilter(filter_title='Location', field_name='location_dim.name', filter_value='None', filter_description=None)
    ],
    comparison_type=models40.ComparisonType.GREATER_THAN,
    cron='0 5 * * *',
    custom_title="New Custom Title to set",
    description=None,
    dashboard_element_id='8',
    destinations=[
        models40.AlertDestination(
            destination_type=models40.DestinationType.EMAIL,
            email_address='test@demo.com',
            action_hub_integration_id=None,
            action_hub_form_params_json=None
        )
    ],
    field=models40.AlertField(title='Location', name='interaction_fact.location_name', filter=[]),
    is_disabled=False,
    is_public=False,
    lookml_dashboard_id=None,
    lookml_link_id=None,
    owner_id='9',
    threshold=1.0,
)

sdk.update_alert(alert_id='30', body=write_alert)

sdk.logout()
```

This raises the following error:
```
Traceback (most recent call last):
  File "looker.py", line 47, in <module>
    sdk.update_alert(alert_id='30', body=write_alert)
  File "/looker-test-i5h05HlC-py3.8/lib/python3.8/site-packages/looker_sdk/sdk/api40/methods.py", line 128, in update_alert
    self.put(
  File "/looker-test-i5h05HlC-py3.8/lib/python3.8/site-packages/looker_sdk/rtl/api_methods.py", line 207, in put
    serialized = self._get_serialized(body)
  File "/looker_sdk/rtl/api_methods.py", line 150, in _get_serialized
    serialized = self.serialize(api_model=body)  # type: ignore
  File "/looker-test-i5h05HlC-py3.8/lib/python3.8/site-packages/looker_sdk/rtl/serialize.py", line 81, in serialize
    return json.dumps(data).encode("utf-8")  # type: ignore
  File "/Python3.framework/Versions/3.8/lib/python3.8/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "/Python3.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/Python3.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/Python3.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type AlertField is not JSON serializable
```

We also see the same issue with `models40.AlertFieldFilter`

Possible workaround that seems to work is to wrap the creation/update of `models40.AlertField` and `models40.AlertFieldFilter` with a `dict` conversion, e.g:

```
filter = [vars(filter) for filter in alert["field"]["filter"]]
field = vars(
    models40.AlertField(
        title=alert["field"]["title"],
        name=alert["field"]["name"],
        filter=filter,
    )
)
```

Please advise if we're using the SDK wrong or if this is indeed a bug in the SDK, since the support agents are unable to answer this.

**Comment by markusos** [comment]
This issue could be a duplicate of https://github.com/looker-open-source/sdk-codegen/issues/1093, but it contains additional information that the problem also affects the `models40.AlertFieldFilter` class serialization.

**Comment by joeldodge79** [comment]
@markusos does the workaround unblock you for now? I'm not sure what's causing this bug yet but I'll dig into it

**Comment by markusos** [comment]
@joeldodge79 yeah, the workaround is fine for now, just wanted the issue tracked so we could get it resolved in a future version of the SDK.
